jstests/fle2/libs/encrypted_client_util.js

```diff
@@ -0,0 +1,190 @@
+     * in the key vault.
+     *
+     * @param {string} name Name of collection
+     * @param {Object} options Create Collection options
+     */
+    createEncryptionCollection(name, options) {
+        assert(options != undefined);
+        assert(options.hasOwnProperty("encryptedFields"));
+        assert(options.encryptedFields.hasOwnProperty("fields"));
+
+        for (let field of options.encryptedFields.fields) {
+            if (!field.hasOwnProperty("keyId")) {
+                let testkeyId = this._keyVault.createKey("local", "ignored");
+                field["keyId"] = testkeyId;
+            }
+        }
+
+        return this._edb.createEncryptedCollection(name, options);
+    }
+
+    /**
+     * Assert the number of documents in the EDC and state collections is correct.
+     *
+     * @param {string} name Name of EDC
+     * @param {number} edc Number of documents in EDC
+     * @param {number} esc Number of documents in ESC
+     * @param {number} ecc Number of documents in ECC
+     * @param {number} ecoc Number of documents in ECOC
+     */
+    assertEncryptedCollectionCounts(name, expectedEdc, expectedEsc, expectedEcc, expectedEcoc) {
+        const ci = this._edb.getCollectionInfos({"name": name})[0];
+
+        const ef = ci.options.encryptedFields;
+
+        const actualEdc = this._edb.getCollection(name).count();
+        assert.eq(actualEdc,
+                  expectedEdc,
+                  `EDC document count is wrong: Actual ${actualEdc} vs Expected ${expectedEdc}`);
+
+        const actualEsc = this._edb.getCollection(ef.escCollection).count();
+        assert.eq(actualEsc,
+                  expectedEsc,
+                  `ESC document count is wrong: Actual ${actualEsc} vs Expected ${expectedEsc}`);
+
+        const actualEcc = this._edb.getCollection(ef.eccCollection).count();
+        assert.eq(actualEcc,
+                  expectedEcc,
+                  `ECC document count is wrong: Actual ${actualEcc} vs Expected ${expectedEcc}`);
+
+        const actualEcoc = this._edb.getCollection(ef.ecocCollection).count();
+        assert.eq(actualEcoc,
+                  expectedEcoc,
+                  `ECOC document count is wrong: Actual ${actualEcoc} vs Expected ${expectedEcoc}`);
+    }
+
+    /**
+     * Get a single document from the collection with the specified query. Ensure it contains the
+     specified fields when decrypted and that does fields are encrypted.
+
+     * @param {string} coll
+     * @param {object} query
+     * @param {object} fields
+     */
+    assertOneEncryptedDocumentFields(coll, query, fields) {
+        let rawDocs = this._db.getCollection(coll).find(query).toArray();
+        assert.eq(rawDocs.length, 1);
+        let encryptedDocs = this._edb.getCollection(coll).find(query).toArray();
+        assert.eq(encryptedDocs.length, 1);
+
+        let rawDoc = rawDocs[0];
+        let encryptedDoc = encryptedDocs[0];
+
+        assert(rawDoc["__safeContent__"] !== undefined);
```

Comment:
So `encryptedDoc` is the document after client-side decryption? and `rawDoc` contains the fields in their encrypted form? If so, the naming confused me at first.